### PR TITLE
Clarify imageName function arguments

### PR DESCRIPTION
**Pull request by djbingham**
https://eaflood.atlassian.net/browse/PSD-40

The same argument name was being used to reference chart, image, project
and service names. Renaming each to reflect is intent makes the library
easier to understand and highlights where two things have been conflated
into a single variable, reducing flexibility in library usage.

### CODE REVIEW

**Comment by djbingham** [comment]
In buildTestImage and runTests, I split the `name` variable into `projectName` and `serviceName` to reflect the two distinct entities referred to. This makes those functions a bit clearer and more flexible. Aside from that, I've just renamed function arguments to clarify their meaning.

File: src/uk/gov/defra/ffc/DefraUtils.groovy [hunk 1]
```diff
@@ -167,32 +166,32 @@ def waitForQualityGateResult(timeoutInMinutes) {
   }
 }
 
-def buildAndPushContainerImage(credentialsId, registry, imageName, tag) {
+def buildAndPushContainerImage(credentialsId, registry, projectName, tag) {
```
Comment (johnwatson484) [inline src/uk/gov/defra/ffc/DefraUtils.groovy]: For the test image, I think `projectName` makes sense as it's used to set the project property.  In this case we are setting the name of the image so `imageName` seems clearer as to our intent.

File: src/uk/gov/defra/ffc/DefraUtils.groovy [hunk 2]
```diff
@@ -167,32 +166,32 @@ def waitForQualityGateResult(timeoutInMinutes) {
...
-    sh "docker tag $imageName $registry/$imageName:$tag"
-    sh "docker push $registry/$imageName:$tag"
+    sh "docker-compose -f docker-compose.yaml build --no-cache"
+    sh "docker tag $projectName $registry/$projectName:$tag"
+    sh "docker push $registry/$projectName:$tag"
   }
 }
 
-def deployChart(credentialsId, registry, imageName, tag, extraCommands) {
+def deployChart(credentialsId, registry, projectName, tag, extraCommands) {
```
Comment (johnwatson484) [inline src/uk/gov/defra/ffc/DefraUtils.groovy]: `helmChart` would be more accurate, like `lintHelm`.

File: src/uk/gov/defra/ffc/DefraUtils.groovy [hunk 3]
```diff
@@ -167,32 +166,32 @@ def waitForQualityGateResult(timeoutInMinutes) {
...
+    def deploymentName = "$projectName-$tag"
     sh "kubectl get namespaces $deploymentName || kubectl create namespace $deploymentName"
-    sh "helm upgrade $deploymentName --install --namespace $deploymentName --atomic ./helm/$imageName --set image=$registry/$imageName:$tag $extraCommands"
+    sh "helm upgrade $deploymentName --install --namespace $deploymentName --atomic ./helm/$projectName --set image=$registry/$projectName:$tag $extraCommands"
   }
 }
 
-def undeployChart(credentialsId, imageName, tag) {
-  def deploymentName = "$imageName-$tag"
+def undeployChart(credentialsId, projectName, tag) {
```
Comment (johnwatson484) [inline src/uk/gov/defra/ffc/DefraUtils.groovy]: `helmChart` name would be more accurate.

File: src/uk/gov/defra/ffc/DefraUtils.groovy [hunk 4]
```diff
@@ -167,32 +166,32 @@ def waitForQualityGateResult(timeoutInMinutes) {
...
+  def deploymentName = "$projectName-$tag"
   echo "removing deployment $deploymentName"
   withKubeConfig([credentialsId: credentialsId]) {
     sh "helm delete --purge $deploymentName || echo error removing deployment $deploymentName"
     sh "kubectl delete namespaces $deploymentName || echo error removing namespace $deploymentName"
   }
 }
 
-def publishChart(registry, imageName, containerTag) {
+def publishChart(registry, projectName, containerTag) {
```
Comment (johnwatson484) [inline src/uk/gov/defra/ffc/DefraUtils.groovy]: if we're shortening the other Helm functions from `containerTag` to `tag` we should do the same here.

File: src/uk/gov/defra/ffc/DefraUtils.groovy [hunk 5]
```diff
@@ -117,23 +117,22 @@ def setGithubStatusFailure(message = '') {
   updateGithubCommitStatus(message, 'FAILURE')
 }
 
-def lintHelm(imageName) {
-  sh "helm lint ./helm/$imageName"
+def lintHelm(projectName) {
```
[the remark this answers is not in the document]
Comment (djbingham) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: Good point, amended.

File: src/uk/gov/defra/ffc/DefraUtils.groovy (hunk 1 above)
Comment (djbingham) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: Agreed, amended.

File: src/uk/gov/defra/ffc/DefraUtils.groovy (hunk 2 above)
Comment (djbingham) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: Changed to `chartName`.

File: src/uk/gov/defra/ffc/DefraUtils.groovy (hunk 3 above)
Comment (djbingham) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: Changed to `chartName`.

File: src/uk/gov/defra/ffc/DefraUtils.groovy (hunk 4 above)
Comment (djbingham) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: I hadn't noticed that change. I've changed it to `tag` here. Also changed `projectName` to `chartName` in this function as well.

**Comment by djbingham** [comment]
> I like how a lot of the dependencies like the build number are now parameterised. I'm not sure on using projectName universally as I think it's a bit unclear what it is.
> 
> The readme also needs updating the match the changes.

Thanks, all good points. You're right about `projectName` not being universally correct, although I thought it was less misleading than `imageName` in most cases. I've updated all the places where you pointed out it's actually being used as either a chart or image name.

I hadn't ever looked at the Readme for this project - gosh it's long! I'll get that updated now.

**Pull request merged**